airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/record_buffer/SerializedBufferingStrategy.java

```diff
@@ -71,10 +65,14 @@ public void addRecord(final AirbyteStreamNameNamespacePair stream, final Airbyte
     if (totalBufferSizeInBytes >= streamBuffer.getMaxTotalBufferSizeInBytes()
         || allBuffers.size() >= streamBuffer.getMaxConcurrentStreamsInBuffer()) {
       flushAll();
+      didFlush = true;
       totalBufferSizeInBytes = 0;
     } else if (streamBuffer.getByteCount() >= streamBuffer.getMaxPerStreamBufferSizeInBytes()) {
       flushWriter(stream, streamBuffer);
```

Comment:
as written now this conditional branch doesn't flush state message. is that okay? i guess the idea is when the buffer is closed they all get flushed, so may be it is not technically incorrect. 

it does mean that we could get behind in flushing the state message, but not the end of the world, i think?

Reply:
Yes, this conditional branch didn't flush the state message because this code is happening only when flushing one of the streams from the local buffers to the remote storage (it is not necessarily flushing all data buffered so far). So since I was not sure if the state was not per-stream or globally, I preferred to "get behind in emitting state message" rather than emit to early

Reply:
okay! makes sense. i'll add a comment to explain why we did this.